technico-v16/app/users/[id]/edit-user/page.tsx

```diff
@@ -0,0 +1,239 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { useParams, useRouter } from "next/navigation";
+import { useAuth } from "@/app/components/AuthContext";
+
+interface User {
+  name: string;
+  surname: string;
+  vatNumber: string;
+  address: string;
+  phoneNumber: string;
+  email: string;
+  password: string;
+}
+
+interface FormErrors {
+    [key: string]: string; // Generic object to hold error messages for each field
+}
+
+const EditUser = () => {
+  const params = useParams();
+  const router = useRouter();
+  const id = params?.id;
+  const {isAuthenticated} = useAuth();
+
+  const [formData, setFormData] = useState<User>({
+    name: "",
+    surname: "",
+    vatNumber: "",
+    address: "",
+    phoneNumber: "",
+    email: "",
+    password: "",
+  });
+
+  const [errors, setErrors] = useState<Partial<User>>({});
+  const [loading, setLoading] = useState(true);
+  const [successMessage, setSuccessMessage] = useState<string | null>(null);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (!id) return;
+    fetchUser();
+  }, [id, isAuthenticated]);
+
+  const fetchUser = async () => {
+    if (!isAuthenticated) return;
+    
+    try {
+      setLoading(true);
+      const response = await fetch(`https://localhost:7118/api/Users/${id}`);
+      if (!response.ok) {
+        throw new Error(`Error ${response.status}: Failed to fetch user data`);
+      }
+      const user = await response.json();
+      setFormData(user);
+    } catch (err: any) {
+      setErrorMessage(err.message || "An error occurred while fetching user data.");
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormData((prev) => ({
+      ...prev,
+      [name]: value,
+    }));
+    setErrors((prev) => ({
+      ...prev,
+      [name]: "",
+    }));
+  };
+
+  const validateForm = (): boolean => {
+    const newErrors: FormErrors = {};
+
+    if (!formData.name) newErrors.name = "Name is required.";
+    if (!formData.surname) newErrors.surname = "Surname is required.";
+    if (!formData.vatNumber || !/^\d{9}$/.test(formData.vatNumber)) {
+        newErrors.vatNumber = "VAT Number must be 9 digits.";
+    }
+    if (!formData.address) newErrors.address = "Address is required.";
+    if (!formData.phoneNumber || !/^\d{10}$/.test(formData.phoneNumber)) {
+        newErrors.phoneNumber = "Phone Number must be 10 digits.";
+    }
+    if (!formData.email || !/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(formData.email)) {
```

Comment:
Good email validation regex altough HTML provides an input component that handles the email inputs similarly.